include/ziti/ziti_tunnel_cbs.h

```diff
@@ -12,6 +12,8 @@ extern "C" {
 typedef struct ziti_io_ctx_s {
     ziti_connection      ziti_conn;
     tunneler_io_context  tnlr_io_ctx;
+    bool ziti_eof;
```

Comment:
It feels like ziti_eof and tnlr_eof should belong to ziti_io_context and tnlr_io_context respecitively.

Reply:
It is easier here because it is the thing that glues them together and has to decide what to do, also `tunneler_io_context` is opaque to `ziti_tunnel_cbs` code and would require accessor or exposed internals